searx/engines/google_images.py

```diff
@@ -61,30 +61,28 @@ def request(query, params):
 
 # get response from search-request
 def response(resp):
+    results = []
+
     dom = html.fromstring(resp.text)
 
-    results = []
-    for element in dom.xpath('//div[@id="search"] //td'):
-        link = element.xpath('./a')[0]
-
-        google_url = urlparse(link.xpath('.//@href')[0])
-        query = parse_qs(google_url.query)
-        source_url = next(iter(query.get('q', [])), None)
-
-        title_parts = element.xpath('./cite//following-sibling::*/text()')
-        title_parts.extend(element.xpath('./cite//following-sibling::text()')[:-1])
-
-        result = {
-            'title': ''.join(title_parts),
-            'content': '',
-            'template': 'images.html',
-            'url': source_url,
-            'img_src': source_url,
-            'thumbnail_src': next(iter(link.xpath('.//img //@src')), None)
-        }
-
-        if not source_url or not result['thumbnail_src']:
+    # parse results
+    for result in dom.xpath('//div[contains(@class, "rg_meta")]/text()'):
+
+        try:
+            metadata = loads(result)
+            format = metadata['ity'] + ' ' + str(metadata['ow']) + 'x' + str(metadata['oh'])
+            source = metadata['st'] + ' (' + metadata['isu'] + ')'
+            results.append({'url': metadata['ru'],
+                            'title': metadata['pt'],
+                            'content': metadata['s'],
+                            'source': source,
+                            'format': format,
+                            'thumbnail_src': metadata['tu'],
+                            'img_src': metadata['ou'],
+                            'template': 'images.html'})
+
+        except:
```

Comment:
Probably `except KeyError`

Reply:
if the JSON can't be decoded it will raise an ```json.JSONDecodeError```  in Python 3 : 
```python
from json import load, JSONDecodeError
...
except (KeyError, JSONDecodeError):
```

in Python 2 it is different (I know).

Anyway the error ignored.

Reply:
Fail-fast and continue FTW, no need to complicate matters by trying to handle these exceptions in some special way since there is no recovery anyway.